### PR TITLE
Add basic authorization check for all controllers

### PR DESCRIPTION
**Pull request by Azuolas3**
Add authorization for all controllers, added authorization option to swagger UI, not allowing non-admin users to add new user with admin role

### CODE REVIEW

File: Controllers/CustomerController.cs [hunk 1]
```diff
@@ -34,16 +37,23 @@ public ActionResult Get(int id)
         [HttpPost]
         public ActionResult Post([FromBody] CustomerCreate body)
         {
+            if (!User.IsInRole("admin") && body.Role == "admin")
+            {
+                return Forbid();
+            }
+
```
Comment (Rytis-Sapka) [inline Controllers/CustomerController.cs:44]: Could you move this to the CustomerService

File: Controllers/AuthController.cs [hunk 2]
```diff
@@ -28,27 +28,30 @@ public AuthController(ICustomersService service, IConfiguration config)
...
+
             if(currentUser == null) 
             {
-                return Unauthorized("Invalid username or password");
+                var unauthorizedResponseJson = new { message = "Invalid username or password" };
+                return Unauthorized(unauthorizedResponseJson);
             }
 
-            return Ok(GenerateToken(currentUser));
+            var responseJson = new { token = GenerateToken(currentUser) };
```
Comment (Rytis-Sapka) [inline Controllers/AuthController.cs:38]: same for this, I think I remember hearing in lectures that controllers should only call a service and the main job of this class is defining input, output, routes, etc...

File: Controllers/CustomerController.cs (hunk 1 above)
Comment (Rytis-Sapka) [reply Controllers/CustomerController.cs:44]: also, perhaps I'm overhinking this a good bit, but I am not sure how people would create their accounts in this case. Would an admin need to create accounts for every single user? (don't think we need to fix this but something to keep in mind)

Comment (Azuolas3) [reply Controllers/CustomerController.cs:44]: No, all users (no auth or authorization required) can create account, but only admins are able to create accounts with admin role

Comment (Rytis-Sapka) [reply Controllers/CustomerController.cs:44]: oh okay, missread it. That's good

File: Controllers/AuthController.cs (hunk 2 above)
Comment (Rytis-Sapka) [reply Controllers/AuthController.cs:38]: sounds good.

**Review by Rytis-Sapka — APPROVED** [review]

**Pull request merged**